egcg_core/rest_communication.py

```diff
@@ -20,6 +22,58 @@ def serialise(queries):
             serialised_queries[k] = json.dumps(v) if type(v) is dict else v
         return serialised_queries
 
+    @staticmethod
+    def get_file_content(file_name):
+        """
+        This method open in binary mode th file provided.
+        :param file_name: path to the file
+        :return: file content
+        """
+        with open(file_name, 'rb') as open_f:
+            fc = open_f.read()
+        return fc
+
+    @staticmethod
+    def _extract_files(json_dict):
+        """
+        This method take a json dict and will search all the values for a tuple where the first element is "file".
+        It extract the value, read the content of the file and create another tuple ready to be used by requests
+        This method will modify the original json_dict and remove the file entry from it.
+        :param json_dict: The json dictionary to search
+        :return: a dict where the keys are the field to update and the value are tuple with (filename, file content, file MIME type)
+        """
+        files = {}
+        for k in json_dict:
+            if isinstance(json_dict[k], tuple) and len(json_dict[k]) > 1 and json_dict[k][0] == 'file':
+                # Construct the 3 parts file upload tuple as requests requires it
+                files[k] = (
+                    json_dict[k][1],                                 # file name
+                    Communicator.get_file_content(json_dict[k][1]),  # file content
+                    mimetypes.guess_type(json_dict[k][1])[0]         # file mime type
+                )
+        if files:
+            # remove the extracted files to the original json
+            for k in files:
+                json_dict.pop(k)
+            return files
+        return None
```

Comment:
No `_extract_files` returns a dict of files per `json_dict` provided.
The purpose of returning `None` explicitly here is to indicate that no files were found in the `json_dict`

```python
json_dict = {'key1': 'val1'}
print(_extract_files(json_dict))
>>> None
json_dict = {'key1': 'val1', 'key2': ('file', '/path/to/file.txt')}
print(_extract_files(json_dict))
>>> {'key2': ('/path/to/file.txt', 'filecontent', 'plain/text')}
```